src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,559 @@
+import {IOptionAsset} from "v2-core/src/interfaces/IOptionAsset.sol";
+import {ISpotFeed} from "v2-core/src/interfaces/ISpotFeed.sol";
+import {IDepositModule} from "../interfaces/IDepositModule.sol";
+import {IWithdrawalModule} from "../interfaces/IWithdrawalModule.sol";
+import {IMatching} from "../interfaces/IMatching.sol";
+
+import {
+  StandardManager, IStandardManager, IVolFeed, IForwardFeed
+} from "v2-core/src/risk-managers/StandardManager.sol";
+import {IRfqModule} from "../interfaces/IRfqModule.sol";
+import "../interfaces/ITradeModule.sol";
+
+/// @title PrincipalProtectedTSA
+/// @notice TSA that accepts any deposited collateral, and sells RFQ spreads on any gained principal.
+contract PrincipalProtectedTSA is BaseOnChainSigningTSA {
+  using IntLib for int;
+  using SafeCast for int;
+  using SafeCast for uint;
+  using DecimalMath for uint;
+  using SignedDecimalMath for int;
+
+  struct PPTSAInitParams {
+    ISpotFeed baseFeed;
+    IDepositModule depositModule;
+    IWithdrawalModule withdrawalModule;
+    IRfqModule rfqModule;
+    IOptionAsset optionAsset;
+  }
+
+  struct PPTSAParams {
+    /// @dev Minimum time before an action is expired
+    uint minSignatureExpiry;
+    /// @dev Maximum time before an action is expired
+    uint maxSignatureExpiry;
+    /// @dev Percentage of spot price that the TSA will buy baseAsset at in the worst case (e.g. 1.02e18)
+    uint worstSpotBuyPrice;
+    /// @dev Percentage of spot price that the TSA will sell baseAsset at in the worst case (e.g. 0.98e18)
+    uint worstSpotSellPrice;
+    /// @dev A factor on how strict to be with preventing too much cash being used in swapping base asset (e.g. 1.01e18)
+    int spotTransactionLeniency;
+    /// @dev The worst difference to vol that is accepted for pricing options (e.g. 0.9e18)
+    uint optionVolSlippageFactor;
+    /// @dev The minimum amount of gain accepted for opening an option position (e.g. 0.01e18)
+    /// POC: This is new
+    uint maxMarkValueToStrikeDiffRatio;
+    /// @dev The maximum amount of gain accepted for opening an option position (e.g. 0.1e18)
+    /// POC: This is new
+    uint minMarkValueToStrikeDiffRatio;
+    /// @dev Lower bound for option expiry
+    uint optionMinTimeToExpiry;
+    /// @dev Upper bound for option expiry
+    uint optionMaxTimeToExpiry;
+    /// @dev Maximum amount of negative cash allowed to be held to open any more option positions. (e.g. -100e18)
+    int optionMaxNegCash;
+    /// @dev Percentage of spot that can be paid as a fee for both spot/options (e.g. 0.01e18)
+    uint feeFactor;
+    /// @dev requirement of distance between two strikes
+    /// POC: This is new
+    int strikeDiff;
+    /// @dev the max tolerance we allow when calculating cost of a trade
+    /// POC: This is new
+    uint maxTotalCostTolerance;
+    /// @dev used as tolerance for how much TVL we can use for RFQ
+    /// POC: This is new
+    uint maxBuyPctOfTVL;
```

Comment:
**Not necessary for PoC**, but down the line I'd consider splitting this struct up and seeing how that affects the contract size. Need to be careful to not go over the size limit. CCTSA was already close.

running `forge build --sizes` you can see what the result looks like. Seems like CCTSA also went over from the last commit or two... Might need to optimise the underlying contracts a bit too... 🤔 

<img width="407" alt="Screenshot 2024-07-03 at 11 59 23 am" src="https://github.com/lyra-finance/v2-matching/assets/11264336/c1c2a604-3fa7-4f65-8c36-e9009b84282d">

Reply:
Changing optimizer_runs in `foundry.toml` to 100 solves it short term... But again, not a concern for PoC...

For a touch of context:

Higher optimizer_runs optimises for gas usage for the end user calling functions, at the cost of contract size

So you basically want to increase it as high as possible without making contracts undeployable (i.e. not exceeding size limit of ~24 kb)